doc/crypto/api/ops/encap.rst

```diff
@@ -0,0 +1,326 @@
+
+    .. param:: ka_alg
+        A encapsulation algorithm: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_ENCAPSULATION(ka_alg)` is true.
+    .. param:: kdf_alg
+        A key derivation algorithm: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_KEY_DERIVATION(kdf_alg)` is true.
+
+    .. return::
+        The corresponding encapsulation and derivation algorithm.
+
+        Unspecified if ``ka_alg`` is not a supported key establlishment algorithm or ``kdf_alg`` is not a supported key derivation algorithm.
+
+    A combined encapsulation algorithm is used in a call to `psa_encapsulate_key()`.
+
+    The component parts of a encapsulation algorithm can be extracted using `PSA_ALG_ENCAPSULATION_GET_BASE()` and `PSA_ALG_ENCAPSULATION_GET_KDF()`.
+
+    .. subsection:: Compatible key types
+
+        The resulting combined encapsulation algorithm is compatible with the same key types as the raw encapsulation algorithm used to construct it.
+
+.. _encapsulation-algorithms:
+
+Encapsulation Algorithms
+------------------------
+.. function:: psa_encapsulate_key
+
+    .. summary::
+        Generate a new key pair and a use that to encapsulate a new symmetric key, emitting it both as a key object and an encapsulation to send to a counter party along with the public key from the ephemeral key pair.
+
+    .. param:: const psa_key_id_t * counterparty_key
+        The identifier for the public key of the peer. You must have previously imported this key using `psa_import_key()`, and specified the key attributes for the public key type corresponding to the type required for the encapsulation, and the usage usage `PSA_KEY_USAGE_ENCAPSULATE_KEY`.
+
+    .. param:: psa_key_id_t * key_pair
+        On success an identifier for the new key pair, which was used to perform the encapsulation.  
+        The key will be the key type specified in the atributes of the counterparty key. The private key will be given the usage `PSA_KEY_USAGE_ENCAPSULATE_KEY`. The public key only will be given the usage `PSA_KEY_USAGE_EXPORT`.  
+        `PSA_KEY_ID_NULL` on failure
+
+    .. param:: uint8_t * ephemeral_public_key
+        Buffer where the ephemeral public key key is to be written, ready to be sent to the counterparty.
+        
+    .. param:: size_t ephemeral_public_key_size
+        Size of the ``ephemeral_public_key`` buffer in bytes.
+        This must be at least :code:`PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE(alg)`.
+        A buffer of at least :code:`PSA_EXPORT_PUBLIC_KEY_MAX_OUTPUT_SIZE`. is guaranteed not to fail due to buffer size for any supported encapsulation algorithm.
+
+    .. param:: psa_algorithm_t alg
+        The ful encapsulation algorithm to use: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_ENCAPSULATION(alg)` is true and :code:`PSA_ALG_IS_RAW_ENCAPSULATION(alg)` is false .
+
+    .. param:: const psa_key_attributes_t * attributes

```

Comment:
I suspect that this should probably have the same constraints/documentation as `psa_key_agreement()`. Depending on the algorithm, the output key/secret may be biased, and the size of the key is determine wholly by the algorithm and key parameters.